producer/writer/message_writer.go

```diff
@@ -290,13 +292,14 @@ func (w *messageWriterImpl) retryBatchWithLock(
 	nowNanos int64,
 ) *list.Element {
 	var (
-		iterated int
-		next     *list.Element
+		iterated   int
+		next       *list.Element
+		retryFound int
 	)
 	for e := start; e != nil; e = next {
 		iterated++
-		if iterated > w.opts.MessageRetryBatchSize() {
-			return e
+		if iterated > w.opts.MessageRetryIterateBatchSize() {
```

Comment:
Do you need two separate batch sizes? Feels like just having `RetryBatchSize` is sufficient.